.meta/transforms/dedupe.toml

```diff
@@ -0,0 +1,43 @@
+[transforms.dedupe]
+title = "Dedupe events"
+allow_you_to_description = "prevent duplicate Events from being outputted"
+beta = false
+common = false
+function_category = "filter"
+input_types = ["log"]
+output_types = ["log"]
+requirements = {}
+
+<%= render("_partials/_component_options.toml", type: "transform", name: "dedupe") %>
+
+[transforms.dedupe.options.fields]
+type = "table"
+description = "Options controlling what fields to match against"
+required = true
+
+[transforms.dedupe.options.fields.children.match]
+type = "[string]"
+common = true
+examples = [["field1", "field2"]]
+default = ["timestamp"]
+description = "The field names considered when deciding if an Event is a duplicate."
+
+[transforms.dedupe.options.fields.children.ignore]
+type = "[string]"
+common = true
+examples = [["field1", "field2"]]
+description = "The field names to ignore when deciding if an Event is a duplicate."
+
+
+[transforms.dedupe.options.cache]
+type = "table"
+description = "Options controlling how we cache recent Events for future duplicate checking."
+required = true
```

Comment:
I think this can be removed?

This generates the following:

```
[transforms.my_transform_id]
  # REQUIRED - General
  type = "dedupe" # must be: "dedupe"
  inputs = ["my-source-id"] # example

  # REQUIRED - Cache
  [transforms.my_transform_id.cache]
    num_entries = 5000 # default

  # REQUIRED - Fields
  [transforms.my_transform_id.fields]
    ignore = ["field1", "field2"] # example, no default
    match = ["timestamp"] # default
```

And I don't think they should need to have that `cache` setting if they just want the default.